pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -632,3 +640,16 @@ func (r *Reconciler) matchControllerRequirement(app *v1beta1.Application) bool {
 	}
 	return true
 }
+
+const (
+	originalAppKey contextKey = "original-app"
+)
+
+func withOriginalApp(ctx context.Context, app *v1beta1.Application) context.Context {
```

Comment:
can we make this function to be a method of `ctrlrec.NewReconcileContext`

Reply:
NewReconcileContext is not only used for application controller. It is used for all KubeVela eco-system controllers.